src/realm/sync/noinst/client_reset_operation.cpp

```diff
@@ -67,83 +51,57 @@ bool ClientResetOperation::is_fresh_path(const std::string& path)
-    m_logger.debug("ClientResetOperation::finalize, realm_path = %1, local_realm_exists = %2, mode = %3",
-                   m_db->get_path(), local_realm_exists, m_mode);
+    auto latest_version = db.get_version_id_of_latest_snapshot();
+    bool local_realm_exists = latest_version.version > 1;
     if (!local_realm_exists) {
+        logger.debug("Local Realm file has never been written to, so skipping client reset.");
         return false;
     }
 
-    REALM_ASSERT_EX(m_db_fresh, m_db->get_path(), m_mode);
-
-    client_reset::LocalVersionIDs local_version_ids;
-    auto always_try_clean_up = util::make_scope_exit([&]() noexcept {
-        clean_up_state();
-    });
-
-    VersionID frozen_before_state_version = m_notify_before ? m_notify_before() : latest_version;
+    VersionID frozen_before_state_version = notify_before ? notify_before() : latest_version;
 
     // If m_notify_after is set, pin the previous state to keep it around.
     TransactionRef previous_state;
-    if (m_notify_after) {
-        previous_state = m_db->start_frozen(frozen_before_state_version);
+    if (notify_after) {
+        previous_state = db.start_frozen(frozen_before_state_version);
     }
     bool did_recover_out = false;
-    local_version_ids = client_reset::perform_client_reset_diff(
-        *m_db, *m_db_fresh, m_salted_file_ident, m_logger, m_mode, m_recovery_is_allowed, &did_recover_out, sub_store,
-        std::move(on_flx_version_complete)); // throws
+    client_reset::perform_client_reset_diff(db, fresh_db, new_file_ident, logger, mode, recovery_is_allowed,
+                                            &did_recover_out, sub_store,
+                                            on_flx_version); // throws
 
-    if (m_notify_after) {
-        m_notify_after(previous_state->get_version_of_current_transaction(), did_recover_out);
+    if (notify_after) {
+        notify_after(previous_state->get_version_of_current_transaction(), did_recover_out);
     }
 
-    m_client_reset_old_version = local_version_ids.old_version;
-    m_client_reset_new_version = local_version_ids.new_version;
-
     return true;
 }
 
-void ClientResetOperation::clean_up_state() noexcept
-{
-    if (m_db_fresh) {
-        std::string path_to_clean = m_db_fresh->get_path();
-        try {
-            // In order to obtain the lock and delete the realm, we first have to close
-            // the Realm. This requires that we are the only remaining ref holder, and
-            // this is expected. Releasing the last ref should release the hold on the
-            // lock file and allow us to clean up.
-            long use_count = m_db_fresh.use_count();
-            REALM_ASSERT_DEBUG_EX(use_count == 1, use_count, path_to_clean);
-            m_db_fresh.reset();
-            // clean up the fresh Realm
-            // we don't mind leaving the fresh lock file around because trying to delete it
-            // here could cause a race if there are multiple resets ongoing
```

Comment:
All this code breaks horribly in any scenario where deleting the lockfile would be a problem. The sync agent claiming on the parent realm is supposed to ensure only one process can be performing a client reset on the realm, and if that's violated very bad things happen, so we don't really need to account for that possiblity.